public/app/plugins/datasource/influxdb/datasource.ts

```diff
@@ -47,6 +47,9 @@ export function InfluxDatasource(instanceSettings, $q, backendSrv, templateSrv)
     // replace templated variables
     allQueries = templateSrv.replace(allQueries, options.scopedVars);
 
+    // replace database name
+    allQueries = allQueries.replace(/\$database/g, this.database);
```

Comment:
but it is added in the request, https://github.com/grafana/grafana/blob/master/public/app/plugins/datasource/influxdb/datasource.ts#L152

https://github.com/grafana/grafana/blob/master/pkg/api/dataproxy.go#L44